addon/decorators/validation/gt.js

```diff
@@ -0,0 +1,38 @@
+import { registerDecorator } from "class-validator";
+import { getOwner } from '@ember/application';
+
+export function Gt(gt, validationOptions) {
+  return function (object, propertyName) {
+    registerDecorator({
+      name: "gt",
+      target: object.constructor,
+      propertyName: propertyName,
+      constraints: [gt],
+      options: validationOptions,
+      validator: {
+        validate(value, { constraints, property }) {
+          if (!constraints || constraints.length < 1) {
+            throw new Error(`You must pass a ${property} value as the first argument to Gte`);
+          }
+          const numValue = Number(value);
+          if (typeof numValue !== 'number' || isNaN(numValue)) {
```

Comment:
It doesn't look like `Number()` can ever return anything but a number or `NaN` so the `typeof` comparison seems superfluous.

Reply:
you're right, good catch. yanked from all existing number comparison validators.